docs/VLSI/Tutorial.rst

```diff
@@ -0,0 +1,137 @@
+.. _tutorial:
+
+ASAP7 Tutorial
+==============
+The ``vlsi`` folder of this repository contains an example HAMMER flow with the SHA-3 accelerator and a dummy hard macro in the ASAP7 PDK. It is intended for use with the Cadence and Mentor tool plugins.
+
+Project Structure
+-----------------
+
+This example gives a suggested file structure and build system. The ``vlsi/`` folder will eventually contain the following files and folders:
+
+* Makefile
+
+  * Integration of Hammer's build system into Chipyard and abstracts away some Hammer commands.
+
+* build
+
+  * Hammer output directory. Can be changed with the ``OBJ_DIR`` variable.
+  * Will contain subdirectories such as ``syn-rundir`` and ``par-rundir`` and the ``inputs.yml`` denoting the top module and input Verilog files.
+
+* bwrc-env.yml
+
+  * An example of tool environment configuration for BWRC affiliates. Replace as necessary for your environment.
+
+* example-vlsi
+
+  * Entry point to Hammer. Contains example placeholders for hooks.
+
+* example.v
+
+  * Verilog wrapper around the accelerator and dummy hard macro.
+
+* example.yml
+
+  * Hammer IR for this tutorial.
+
+* extra_libraries
+
+  * Contains collateral for the dummy hard macro.
+
+* generated-src
+
+  * All of the elaborated Chisel and FIRRTL.
+
+* hammer, hammer-<vendor>-plugins, hammer-<tech>-plugin
+
+  * Core, tool, tech repositories.
+
+Prerequisites
+-------------
+
+* Python 3.4+
+* numpy and gdspy packages
+* Genus, Innovus, and Calibre licenses
+* For ASAP7 specifically:
+
+  * Download the `ASAP7 PDK <http://asap.asu.edu/asap/>`__ tarball and do not extract it
+  * If you have additional ASAP7 hard macros, their LEF & GDS need to be 4x upscaled @ 4000 DBU precision
+
+Initial Setup
+-------------
+In the Chipyard root, run:
+
+.. code-block:: shell
+
+    ``./scripts/init-vlsi.sh asap7`` 
+    
+to pull the HAMMER & plugin submodules. Note that for technologies other than ``asap7``, the tech submodule must be added in the ``vlsi`` folder first.
+
+Pull the Hammer environment into the shell:
+
+.. code-block:: shell
+
+    cd vlsi
+    export HAMMER_HOME=$PWD/hammer
+    source $HAMMER_HOME/sourceme.sh
+
+Building the Design
+-------------------
+To elaborate the Sha3RocketConfig (Rocketchip w/ the accelerator) and set up all prerequisites for the build system to push just the accelerator + hard macro through the flow:
+
+.. code-block:: shell
+
+    export MACROCOMPILER_MODE='--mode synflops'
+    export CONFIG=Sha3RocketConfig
+    export VLSI_TOP=Sha3AccelwBB
+    make buildfile
+
+Note that because the ASAP7 process does not yet have a memory compiler, flip-flop arrays are used instead.
```

Comment:
If might be worth structuring this in a way that exlains why each environemnt variable is set:

The `MACROCOMPILER_MODE='--mode synflops' environment variable is required since the ASAP7 process does not yet have a memory compiler. Therefore, flip-flop arrays (`synflops`) are used instead.

The `CONFIG=Sha3RocketConfig` selects the target  generator config in the same manner as the rest of the chipyard framework.

The `VLSI_TOP=Sha3AccelwBB` environment variable indicates to hammer that we are interested in the physical design of only the SHA3 accelerator block (rather than the entire RTL SoC).